tests/phpunit/tests/functions/wpUniquePrefixedId.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/**
+ * Test cases for the `wp_unique_prefixed_id()` function.
+ *
+ * @package WordPress\UnitTests
+ *
+ * @since 6.4.0
+ *
+ * @group functions.php
+ * @covers ::wp_unique_prefixed_id
+ * @runTestsInSeparateProcesses
+ * @preserveGlobalState disabled
+ */
+class Tests_Functions_WpUniquePrefixedId extends WP_UnitTestCase {
+
+	/**
+	 * Tests that the expected unique prefixed IDs are created.
+	 *
+	 * @ticket 59681
+	 *
+	 * @dataProvider data_should_create_unique_prefixed_ids
+	 *
+	 * @param mixed $prefix   The prefix.
+	 * @param array $expected The next two expected IDs.
+	 */
+	public function test_should_create_unique_prefixed_ids( $prefix, $expected ) {
+		$id1 = wp_unique_prefixed_id( $prefix );
+		$id2 = wp_unique_prefixed_id( $prefix );
+
+		$this->assertNotSame( $id1, $id2, 'The IDs are not unique.' );
+		$this->assertSame( $expected, array( $id1, $id2 ), 'The IDs did not match the expected values.' );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
+	 */
+	public function data_should_create_unique_prefixed_ids() {
+		return array(
+			'prefix as empty string'       => array(
+				'prefix'   => '',
+				'expected' => array( '1', '2' ),
+			),
+			'prefix as (string) "0"'       => array(
+				'prefix'   => '0',
+				'expected' => array( '01', '02' ),
+			),
+			'prefix as string'             => array(
+				'prefix'   => 'test',
+				'expected' => array( 'test1', 'test2' ),
+			),
+			'prefix as string with spaces' => array(
+				'prefix'   => '   ',
+				'expected' => array( '   1', '   2' ),
+			),
+			'prefix as (string) "1"'       => array(
+				'prefix'   => '1',
+				'expected' => array( '11', '12' ),
+			),
+			'prefix as a (string) "."'     => array(
+				'prefix'   => '.',
+				'expected' => array( '.1', '.2' ),
+			),
+			'prefix as a block name'       => array(
+				'prefix'   => 'core/list-item',
+				'expected' => array( 'core/list-item1', 'core/list-item2' ),
+			),
+		);
+	}
+
+	/**
+	 * @ticket 59681
+	 *
+	 * @dataProvider data_should_raise_notice_and_use_empty_string_prefix_when_nonstring_given
+	 *
+	 * @param mixed  $non_string_prefix Non-string prefix.
+	 * @param string $expected_message  Expected notice message.
+	 * @param string $expected_value    Expected unique ID.
+	 */
+	public function test_should_raise_notice_and_use_empty_string_prefix_when_nonstring_given( $non_string_prefix, $expected_message, $expected_value ) {
```

Comment:
I think this can be a post commit cleanup, but If this test is ever run before `test_should_create_unique_prefixed_ids` then they will fail.

Reply:
Yes, the order of the tests within the class does currently matter. The tests in this class runs in a separate process to avoid a situation where the static`$id_counters` already has values in it. 

To handle the test run order within the class, could change that to run each test in a separate process. By doing so, the static `$id_counters` will always be reset to an empty array when each test starts.

Reply:
Commit 5aac2049e9d55dd21d0c589b11108cc839fde67c runs each test dataset in a separate process to avoid test order concerns,

Reply:
I also tweaked the unhappy path tests to ensure the IDs remain unique f4f99fc242a565eda8548a678cdd8423faf13e65

How? By allowing each dataset to determine how many unique IDs to generate.

Why? Each of these non-string prefix datasets would have the same ID, as the prefix is omitted (empty string). One more check just to make sure.